src/main/java/org/candlepin/async/tasks/AnonymousRegisterAndEntitleJob.java

```diff
@@ -0,0 +1,208 @@
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.ArgumentConversionException;
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.JobArguments;
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobConstraints;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.service.AnonymousOwner;
+import org.candlepin.service.CloudProvider;
+import org.candlepin.service.CloudRegistrationAdapter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+public class AnonymousRegisterAndEntitleJob implements AsyncJob {
+
+    private static final Logger log = LoggerFactory.getLogger(AnonymousRegisterAndEntitleJob.class);
+
+    public static final String JOB_KEY = "AnonymousRegisterJob";
+    public static final String JOB_NAME = "Anonymous Register Job";
+
+    protected static final String ACCOUNT_ID = "account_id";
+    protected static final String OFFERING_ID = "offering_id";
+    protected static final String OWNER_KEY = "owner_key";
+    protected static final String CLOUD_PROVIDER = "cloud_provider";
+
+    private final CloudRegistrationAdapter cloudAdapter;
+
+
+    @Inject
+    public AnonymousRegisterAndEntitleJob(CloudRegistrationAdapter cloudAdapter) {
+
+        this.cloudAdapter = Objects.requireNonNull(cloudAdapter);
+    }
+
+    /**
+     * Executes this job.
+     *
+     * @param context the execution context for the job, containing the arguments
+     *                and other context-specific data
+     * @throws JobExecutionException if an exception occurs during job execution
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String accountId = args.getAsString(ACCOUNT_ID);
+        String offeringId = args.getAsString(OFFERING_ID);
+        String ownerKey = args.getAsString(OWNER_KEY);
+        CloudProvider cloudProviderShortName = args.getAs(CLOUD_PROVIDER, CloudProvider.class);
+
+        log.debug("Starting isAnonymous register and/or entitle job");
+
+        AnonymousOwner owner;
+        try {
+            owner = this.cloudAdapter.createAndEntitleOrg(
+                accountId, offeringId, ownerKey, cloudProviderShortName);
+        }
+        catch (Exception e) {
```

Comment:
Probably shouldn't be catching all exceptions here, and the scoping of the `owner` variable should be tightened up to the try-catch block.

Reply:
done